hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRenameRequest.java

```diff
@@ -0,0 +1,203 @@
+    .RenameKeyResponse;
+import org.apache.hadoop.ozone.security.acl.IAccessAuthorizer;
+import org.apache.hadoop.ozone.security.acl.OzoneObj;
+import org.apache.hadoop.util.Time;
+import org.apache.hadoop.utils.db.Table;
+import org.apache.hadoop.utils.db.cache.CacheKey;
+import org.apache.hadoop.utils.db.cache.CacheValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.apache.hadoop.ozone.om.exceptions.OMException.ResultCodes.KEY_NOT_FOUND;
+
+/**
+ * Handles rename key request.
+ */
+public class OMKeyRenameRequest extends OMClientRequest
+    implements OMKeyRequest {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMKeyRenameRequest.class);
+
+  public OMKeyRenameRequest(OMRequest omRequest) {
+    super(omRequest);
+  }
+
+  @Override
+  public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
+
+    RenameKeyRequest renameKeyRequest = getOmRequest().getRenameKeyRequest();
+    Preconditions.checkNotNull(renameKeyRequest);
+
+    // Set modification time.
+    KeyArgs.Builder newKeyArgs = renameKeyRequest.getKeyArgs().toBuilder()
+            .setModificationTime(Time.now());
+
+    return getOmRequest().toBuilder()
+        .setRenameKeyRequest(renameKeyRequest.toBuilder()
+            .setKeyArgs(newKeyArgs)).setUserInfo(getUserInfo()).build();
+
+  }
+
+  @Override
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
+      long transactionLogIndex) {
+
+    RenameKeyRequest renameKeyRequest = getOmRequest().getRenameKeyRequest();
+    Preconditions.checkNotNull(renameKeyRequest);
+
+    OzoneManagerProtocolProtos.KeyArgs renameKeyArgs =
+        renameKeyRequest.getKeyArgs();
+
+    String volumeName = renameKeyArgs.getVolumeName();
+    String bucketName = renameKeyArgs.getBucketName();
+    String fromKeyName = renameKeyArgs.getKeyName();
+    String toKeyName = renameKeyRequest.getToKeyName();
+
+    OMMetrics omMetrics = ozoneManager.getMetrics();
+    omMetrics.incNumKeyRenames();
+
+    AuditLogger auditLogger = ozoneManager.getAuditLogger();
+
+    Map<String, String> auditMap = buildKeyArgsAuditMap(renameKeyArgs);
+
+    OzoneManagerProtocolProtos.OMResponse.Builder omResponse =
+        OzoneManagerProtocolProtos.OMResponse.newBuilder().setCmdType(
+            OzoneManagerProtocolProtos.Type.CommitKey).setStatus(
+            OzoneManagerProtocolProtos.Status.OK).setSuccess(true);
+
+    try {
+      if (toKeyName.length() == 0 || fromKeyName.length() == 0) {
+        throw new OMException("Key name is empty",
+            OMException.ResultCodes.INVALID_KEY_NAME);
+      }
+      // check Acl
+      if (ozoneManager.getAclsEnabled()) {
+        checkAcls(ozoneManager, OzoneObj.ResourceType.KEY,
+            OzoneObj.StoreType.OZONE, IAccessAuthorizer.ACLType.WRITE,
+            volumeName, bucketName, fromKeyName);
```

Comment:
There is some repetitive code between all the Requests (ACL checks and audit logging). They can probably be combined in one place. We don't need to do this in this Jira. Do you want to open a new Jira for that to keep track?

Reply:
Already we have a method checkAcl in OmRequest class. If we want to reduce the number of lines of code. And remaining lines of code are specific to each request. And same for auditLogging, in super class we have a method, we call that. Let me know if I am missing something here.